src/compiler/sys.ts

```diff
@@ -361,6 +386,45 @@ namespace ts {
         }
     }
 
+    function createFixedChunkSizePollingWatchFile(host: {
+        getModifiedTime: NonNullable<System["getModifiedTime"]>;
+        setTimeout: NonNullable<System["setTimeout"]>;
+    }): HostWatchFile {
+        const watchedFiles: (WatchedFileWithIsClosed | undefined)[] = [];
+        let pollIndex = 0;
+        let pollScheduled: any;
+        return watchFile;
+
+        function watchFile(fileName: string, callback: FileWatcherCallback): FileWatcher {
+            const file: WatchedFileWithIsClosed = {
+                fileName,
+                callback,
+                mtime: getModifiedTime(host, fileName)
+            };
+            watchedFiles.push(file);
+            scheduleNextPoll();
+            return {
+                close: () => {
+                    file.isClosed = true;
+                    // Remove from watchedFiles
+                    unorderedRemoveItem(watchedFiles, file);
+                }
+            };
+        }
+
+        function pollQueue() {
+            pollScheduled = undefined;
+            // Max visit would be all elements of the queue

```

Comment:
I think this comment ended up inside the helper function and is unnecessary here?